### ISSUE TITLE
Support files have different context to spec files

### ISSUE THREAD

**Issue by WORMSS** [cypress-io/cypress#19011]
### Current behavior

There seems to be different context for support files and spec files which was not expected and was causing massive headaches for us trying to work out why things that had been setup were suddenly not.

The files below 'ACTUALLY' run as such
```
LOG: support/index before() 0.5644876764
LOG: support/index before() 0.5644876764 AGAIN
LOG: specs/SomeFancyTest.spec before() 0.8976413461
LOG: specs/SomeFancyTest.spec before() 0.8976413461 AGAIN
LOG: specs/SomeFancyTest.spec it() 0.8976413461
LOG: specs/SomeFancyTest.spec it() 0.8976413461 AGAIN
```

So anything set in the support before() is meaningless

### Desired behavior

support files have the same context as the spec files.

for instance, the code below 'SHOULD' run as such
```
LOG: support/index before() 0.5644876764
LOG: support/index before() 0.5644876764 AGAIN
LOG: specs/SomeFancyTest.spec before() 0.5644876764
LOG: specs/SomeFancyTest.spec before() 0.5644876764 AGAIN
LOG: specs/SomeFancyTest.spec it() 0.5644876764
LOG: specs/SomeFancyTest.spec it() 0.5644876764 AGAIN
```

### TestStuff.js
```js
export class TestingStuff {
  static getId() {
    return this.id ?? (this.id = Math.random().toString());
  }
}
```
### support/index.js
```js
import { TestingStuff } from 'TestingStuff'
before(() => {
  cy.log(`support/index before() ${TestingStuff.getId()}`);
  cy.log(`support/index before() ${TestingStuff.getId()} AGAIN`);
});
```

### specs/SomeFancyTest.spec.js
```js
import { TestingStuff } from 'TestingStuff'
describe('I love testing', () => {
  before(() => {
    cy.log(`specs/SomeFancyTest.spec before() ${TestingStuff.getId()}`);
    cy.log(`specs/SomeFancyTest.spec before() ${TestingStuff.getId()} AGAIN`);
  });
  it('should be good', () => {
    cy.log(`specs/SomeFancyTest.spec it() ${TestingStuff.getId()}`);
    cy.log(`specs/SomeFancyTest.spec it() ${TestingStuff.getId()} AGAIN`);
  });
});
```
### Cypress Version

8.3.1

### Other

Cypress package version: 8.3.1
Cypress binary version: 8.3.1
Electron version: 13.2.0     
Bundled Node version:        
14.16.0

**Comment by chrisbreiding** [comment]
This is expected behavior. The support file generates a separate file from the one generated by your spec. I seem to recall some discussion in an issue about pulling the support file into the spec so they're in the same generated file. I wish I could find it, as I remember there being pros and cons listed for both ways, but I unfortunately haven't been able to. 

In any case, we don't plan to change this behavior any time soon.

You can achieve what you're aiming for in a couple ways:

### Option 1: Use globals

support/index.js

```js
import { TestingStuff } from 'TestingStuff'

window.TestingStuff = TestingStuff

before(() => {
  cy.log(`support/index before() ${TestingStuff.getId()}`);
  cy.log(`support/index before() ${TestingStuff.getId()} AGAIN`);
});
```

spec.js

```js
// no need to import TestingStuff, it's available as a global
describe('I love testing', () => {
  before(() => {
    cy.log(`specs/SomeFancyTest.spec before() ${window.TestingStuff.getId()}`);
    cy.log(`specs/SomeFancyTest.spec before() ${window.TestingStuff.getId()} AGAIN`);
  });
  it('should be good', () => {
    cy.log(`specs/SomeFancyTest.spec it() ${window.TestingStuff.getId()}`);
    cy.log(`specs/SomeFancyTest.spec it() ${window.TestingStuff.getId()} AGAIN`);
  });
});
```

### Option 2: Use imported file for before hook

Don't really need to use the support file for this option, since it's a convenience to save you from needing to import into every spec file. Since the spec file already imports a file, you can just use that file for setting up hooks.

TestingStuff.js

```js
export class TestingStuff {
  static getId() {
    return this.id ?? (this.id = Math.random().toString());
  }
}

before(() => {
  cy.log(`support/index before() ${TestingStuff.getId()}`);
  cy.log(`support/index before() ${TestingStuff.getId()} AGAIN`);
});
```

spec.js

```js
import { TestingStuff } from 'TestingStuff'

describe('I love testing', () => {
  before(() => {
    cy.log(`specs/SomeFancyTest.spec before() ${TestingStuff.getId()}`);
    cy.log(`specs/SomeFancyTest.spec before() ${TestingStuff.getId()} AGAIN`);
  });
  it('should be good', () => {
    cy.log(`specs/SomeFancyTest.spec it() ${TestingStuff.getId()}`);
    cy.log(`specs/SomeFancyTest.spec it() ${TestingStuff.getId()} AGAIN`);
  });
});
```

**Comment by cmcnicholas** [comment]
This is a shame as the usefulness of that file was like jest or mocha setup files. You could initialise something in a consistent way e.g. importing dayjs plugins or locales. The usefulness of this support/index file is severely reduced in a TypeScript environment as you shouldn't just be elevating this stuff to globals.

**Comment by WORMSS** [comment]
We went for **Option 3**
Just run a before in every single spec file that does the setup, and just accept it will happen multiple times.
We already have a a shared function for fixtures that has its own before() methods. So for quick-fix we just hooked into that.

I would like to ask that this behaviour be made very much more obvious in the docs somewhere.

This was one of those problems that got discovered by one person, ended up having to report it internally, and then me spending a good amount of time going completely down the wrong track until I started doing dodgy nonsense like my example above to prove to myself where the global contexts were different.
